File: app/models/RelationDescription.java
/*
 * RelationDescription.java
 */


package models;

import javax.persistence.Entity;
import javax.persistence.Table;
import javax.persistence.Id;
import javax.persistence.Column;
import javax.persistence.GeneratedValue;
import javax.persistence.ManyToOne;
import javax.persistence.MapsId;
import javax.persistence.JoinColumn;
import javax.persistence.Basic;

import play.db.ebean.Model;


/**
 * RelationDescription
 */
@Entity
@Table(name = "tbl_relation_descriptions")
public class RelationDescription extends Model {

    @Id
    @Column(name = "relation_description_id")
    @GeneratedValue
    public Integer relation_description_id;

    @ManyToOne
    @MapsId
    @JoinColumn(name = "relation_id")
    Relation _relation;

    @Basic
    @Column(name = "description", length = 512)
    public String description;

    @Basic
    @Column(name = "type", length = 512)
    public String type;

    @Basic
    @Column(name = "lang", length = 64)
    public String lang;

    @Basic
    @Column(name = "url", length = 512)
    public String url;
}

File: app/models/dto/RelationDescription.scala
package models.dto

import play.api.libs.json.Json

/** RelationDescription DTO (Data Transfer Object)
  *
  * Entity class storing rows of table RelationDescriptions.
  *
  * @param relationDescriptionId Database column RELATION_DESCRIPTION_ID PrimaryKey
  * @param relationId Database column RELATION_ID
  * @param description Database column DESCRIPTION
  * @param `type` Database column TYPE
  * @param lang Database column LANG
  * @param url Database column URL
  */
case class RelationDescription(
    relationDescriptionId: Option[Int],
    relationId: Int,
    description: Option[String],
    `type`: Option[String],
    lang: Option[String],
    url: Option[String])

object RelationDescription {
  implicit val relationDescriptionReads = Json.reads[RelationDescription]
  implicit val relationDescriptionWrites = Json.writes[RelationDescription]
}
